fairlearn/metrics/_metrics_engine.py

```diff
@@ -3,104 +3,124 @@
 
 import numpy as np
 
-from ._group_metric_result import GroupMetricResult
 from ._input_manipulations import _convert_to_ndarray_and_squeeze
+from sklearn.utils import Bunch
 
 _MESSAGE_SIZE_MISMATCH = "Array {0} is not the same size as {1}"
 
+# Parameters to metrics that should be split according to sensitive features
+_SPLITTABLE_METRIC_PARAMS = {"sample_weight"}
 
-def metric_by_group(metric_function,
-                    y_true, y_pred, group_membership,
-                    sample_weight=None,
-                    **kwargs):
+
+def group_summary(metric_function, y_true, y_pred,
+                  sensitive_features, **metric_params):
```

Comment:
I thought `**kwargs` was conventional? I can see the sense in renaming it, but I wouldn't want to confuse future readers.

Reply:
I don't think that `**kwargs` is a major dogma. In this specific instance, calling it with what it contains seems clearer to me, but I don't have a strong opinion on this.

Reply:
Perhaps `metric_function_params` ? I was just surprised, but I'm still on a Python learning curve.

Reply:
I've never seen it called anything but `kwargs` but I don't see why not...

Reply:
I'll rename it to `**metric_function_kwargs` to make this clearer?

Reply:
FWIW we call them `**fit_params` in sklearn.

Reply:
I'm OK with `**metric_func_params` then

Reply:
done.